src/rpc/blockchain.cpp

```diff
@@ -792,6 +745,43 @@ static RPCHelpMan getmempoolentry()
     };
 }
 
+static RPCHelpMan getblockfrompeer()
+{
+    return RPCHelpMan{"getblockfrompeer",
+                "\nAttempt to fetch block from a given peer.\n",
```

Comment:
Perhaps document that this node must already have the block header in order to fetch the block (or relax that requirement?)

Reply:
By "this node" you mean the node itself? What happens when you fetch a block that you don't have the header for?

Reply:
I see `MarkBlockAsInFlight` can handle a `nullprtr` index. It might indeed be simpler to drop this requirements, as well as the `index.nHeight > node_sync_height` check.

Reply:
Sorry, yes I meant "we" as "this node" (the node that you're running this command on)